Swiftui table (#2176)
Codecademy/docs: content/swiftui/concepts/views/terms/table/table.md

---
Title: `Table`
Description: `A container that presents rows of data arranged in one or more columns, optionally providing the ability to select one or more members.`
Subjects: Mobile Development, iOS application, SwiftUI
Tags: SwiftUI, iOS, table, view

---

> A **Table** is a container that presents rows of data arranged in one or more columns, optionally providing the ability to select one or more members.

Reviewer: can we mention that it's a structure in terms of swiftui

> Tables are commonly created from collections of data.
> If there are more rows than can fit in the available space, the table provides vertical scrolling automatically. On macOS, the table also provides horizontal scrolling if there are more columns than can fit in the width of the view.

Reviewer: ```suggestion
Tables are commonly created from collections of data. If there are more rows than what can fit in the available space, the table provides vertical scrolling automatically. On macOS, the table also provides horizontal scrolling if there are more columns than what can fit in the width of the view.
```

> >struct Product: Identifiable {
>  let id: UUID()
>  let name: String
>  let color: String
>  let price: Double
> }
> 
> >@State private var products = [
> 	Product(name: "Pen", color: "Black", price: 2.99),
> 	Product(name: "Eraser", color: "White", price: 1.49),
> 	Product(name: Pencil, color: "Green", price: 0.99)
> 	]
> 
> >struct ProductTable: View {
> 	var body: some View {
> 		Table (products) {
> 			TableColumn ("Product Name", value: \.name)
> 			TableColumn("Product Color", value: \.color)
> 			TableColumn("Product Price", value: \.price) 
> 		}
> 	 }
> 	}

Reviewer: ```suggestion
struct Product: Identifiable {
    let id: UUID()
    let name: String
    let color: String
    let price: Double
}

@State private var products = [
    Product(name: "Pen", color: "Black", price: 2.99),
    Product(name: "Eraser", color: "White", price: 1.49),
    Product(name: Pencil, color: "Green", price: 0.99)
]

struct ProductTable: View {
    var body: some View {
        Table (products) {
            TableColumn ("Product Name", value: \.name)
            TableColumn("Product Color", value: \.color)
            TableColumn("Product Price", value: \.price) 
        }
    }
}
```

Reviewer: and like @SSwiniarski mentioned, place this in a code block. To open the block use three backticks and swift: "```swift" then close with three backticks

Reviewer: This code block would also need some explanation. If you want to simplify the example in terms of complexity, feel free to do so and we can reevaluate

	
 

Tables are interacted with by selecting one or more rows and then performing actions from the toolbar or a context menu.

To make rows of a table selectable, provide a binding to a selection variable
* Binding to a single instance of the table data’s id type creates a single-selection table
* Binding to a Set creates a table that supports multiple selections

## Styling Tables

Use tableStyle (_:) modifier to set a TableStyle for all the tables within view
* AutomaticTableStyle
* InsetTableStyle

> * BorderedTableStyle

Reviewer: What do these represent? Can you preface this in the sentence above and go into the list using colon at the end of the sentence like this:

- item1: Significance
- item2: If any

---

> Title: SwiftUI table

Reviewer 2: ```suggestion
Title: `Table`
```

> Description: Table view in SwiftUI

Reviewer 2: ```suggestion
Description: `A container that presents rows of data arranged in one or more columns, optionally providing the ability to select one or more members.`
```

Subjects: Mobile Development, iOS application, SwiftUI
Tags: SwiftUI, iOS, table, view

---

# SWIFTUI - LIST

> ## **A container that presents rows of data arranged in one or more columns, optionally providing the ability to select one or more members**

Reviewer 2: ```suggestion
A **Table** is a container that presents rows of data arranged in one or more columns, optionally providing the ability to select one or more members.
```
This is not prper use for a header, and we only bold the first instance of the entry name.

You commonly create tables from collections of data.
If there are more rows than can fit in the available space,  table provides vertical scrolling automatically (on macOS, the table also provides horizontal scrolling if there are more columns than can fit in the width of the view)

> >struct Product: Identifiable {

Reviewer 2: Enclose the code example in a Markdown code block. Precede the code block with an `## Example` header and a sentence describing what the example does.  It would be good to provide an image showing what the output of this code looks like.

 let id: UUID()
 let name: String
 let color: String
 let price: Double
}

>@State private var products = [
	Product(name: "Pen", color: "Black", price: 2.99),
	Product(name: "Eraser", color: "White", price: 1.49),
	Product(name: Pencil, color: "Green", price: 0.99)
	]

>struct ProductTable: View {
	var body: some View {
		Table (products) {
			TableColumn ("Product Name", value: \.name)
			TableColumn("Product Color", value: \.color)
			TableColumn("Product Price", value: \.price) 
		}
	 }
	}
	
 

> ### You interact with the table by selecting one or more rows and then performing actions from the toolbar or a context menu.

Reviewer 2: ```suggestion
Tables are interacted with by selecting one or more rows and then performing actions from the toolbar or a context menu.
```
This isn't appropriate for a header. Headers are 5 words or less, in title case, and have no ending punctuation.

> To make rows of a table selectable, provide a binding to a selection variable

Reviewer 2: Can we have a code example of this?

* Binding to a single instance of the table data’s id type creates a single-selection table
* Binding to a Set creates a table that supports multiple selections

> ## Styling tabels

Reviewer 2: ```suggestion
## Styling Tables
```

> Use tableStyle (_:) modifier to set a TableStyle for all the tables within view

Reviewer 2: Can we have a code example of this?

Reviewer: ditto on this

* AutmaticTableStyle
* InsetTableStyle
* BorderedTableStyle

> 

Reviewer 2: Remove all the extra blank lines at the end of this entry.
